src/perfoutputwidgetkonsole.cpp

```diff
@@ -0,0 +1,166 @@
+    : PerfOutputWidget(parent)
+    , m_konsolePart(part)
+{
+    auto layout = new QVBoxLayout(this);
+    setLayout(layout);
+
+    addPartToLayout();
+}
+
+PerfOutputWidgetKonsole::~PerfOutputWidgetKonsole() = default;
+
+PerfOutputWidgetKonsole* PerfOutputWidgetKonsole::create(QWidget* parent)
+{
+    m_tailPath = findTail();
+    if (m_tailPath.isEmpty()) {
+        return nullptr;
+    }
+
+    auto part = createPart();
+    if (!part) {
+        return nullptr;
+    }
+
+    const auto terminalInterface = qobject_cast<TerminalInterface*>(part);
+    if (!terminalInterface) {
+        qWarning("konsole kpart doesn't implement terminal interface");
+        delete part;
+        return nullptr;
+    }
+
+    return new PerfOutputWidgetKonsole(part, parent);
+}
+
+bool PerfOutputWidgetKonsole::eventFilter(QObject* watched, QEvent* event)
+{
+    Q_UNUSED(watched);
+    if (event->type() == QEvent::KeyPress) {
+        auto keyEvent = static_cast<QKeyEvent*>(event);
+
+        // ignore ctrl + c
+        if (keyEvent->key() == Qt::Key_C && keyEvent->modifiers() == Qt::ControlModifier) {
+            return true;
+        } else if (keyEvent->key() == Qt::Key_S && keyEvent->modifiers() == Qt::ControlModifier) {
+            // ignore ctrl + s (stop output)
+            return true;
+        }
+
+        if (!m_inputEnabled) {
+            // eat all key events if input is disabled
+            return true;
+        }
+    } else if (event->type() == QEvent::MouseButtonPress) {
+        auto mouseEvent = static_cast<QMouseEvent*>(event);
+
+        // prevent the user from opening the context menu
+        if (mouseEvent->button() == Qt::MouseButton::RightButton) {
+            return true;
+        }
+    }
+    return false;
+}
+
+void PerfOutputWidgetKonsole::addOutput(const QString& output)
+{
+    m_konsoleFile->write(output.toUtf8());
+    m_konsoleFile->flush();
+}
+
+void PerfOutputWidgetKonsole::clear()
+{
+    if (m_konsolePart) {
+        m_konsolePart->deleteLater();
+    }
+
+    m_konsolePart = createPart();
+
+    if (m_konsoleFile) {
+        m_konsoleFile->deleteLater();
+    }
+
+    addPartToLayout();
+}
+
+void PerfOutputWidgetKonsole::enableInput(bool enable)
+{
+    if (enable != m_inputEnabled) {
+        m_inputEnabled = enable;
+    }
+}
+
+void PerfOutputWidgetKonsole::setInputVisible(bool visible)
+{
+    Q_UNUSED(visible);
+}
+
+QString PerfOutputWidgetKonsole::findTail()
+{
+    const auto tail = QStandardPaths::findExecutable(QStringLiteral("tail"));
+
+    return tail;
+}
+
+KParts::ReadOnlyPart* PerfOutputWidgetKonsole::createPart()
+{
+    KService::Ptr service = KService::serviceByDesktopName(QStringLiteral("konsolepart"));
+
+    if (!service) {
+        return nullptr;
+    }
+    return service->createInstance<KParts::ReadOnlyPart>();
+}
+
+void PerfOutputWidgetKonsole::addPartToLayout()
+{
+    m_konsoleFile = new QTemporaryFile(this);
```

Comment:
you could parent this to the konsole part, then you wouldn't have to manually delete it in the `::clear` method above